src/main/java/net/dv8tion/jda/api/entities/Guild.java

```diff
@@ -408,6 +408,62 @@ default Member getMemberById(long userId)
         return getMemberCache().getElementById(userId);
     }
 
+    /**
+     * Searches for a {@link net.dv8tion.jda.api.entities.Member} that has the matching Discord Tag.
+     * <br>Format has to be in the form {@code Username#Discriminator} where the
+     * username must be between 2 and 32 characters (inclusive) matching the exact casing and the discriminator
+     * must be exactly 4 digits.
+     * <br>This does not check the {@link net.dv8tion.jda.api.entities.Member#getNickname() nickname} of the member
+     * but the username.
+     *
+     * <p>This only checks users that are in this guild. If a user exists
+     * with the tag that is not available in the {@link #getMemberCache() Member-Cache} it will not be detected.
+     * <br>Currently Discord does not offer a way to retrieve a user by their discord tag.
+     *
+     * @param  tag
+     *         The Discord Tag in the format {@code Username#Discriminator}
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided tag is null or not in the described format
+     *
+     * @return The {@link net.dv8tion.jda.api.entities.Member} for the discord tag or null if no member has the provided tag
+     *
+     * @see    net.dv8tion.jda.api.JDA#getUserByTag(String)
+     */
+    default Member getMemberByTag(String tag)
+    {
+        User user = getJDA().getUserByTag(tag);
+        return getMember(user);
```

Comment:
This will NPE if `getUserByTag` returns `null`

Also, we should probably add an `Check.notNull` in `getMember(user)`